website/source/docs/auth/aws-iam.html.md

```diff
@@ -0,0 +1,606 @@
+---
+layout: "docs"
+page_title: "Auth Backend: AWS-IAM"
+sidebar_current: "docs-auth-aws-iam"
+description: |-
+  The aws-iam backend allows automated authentication AWS IAM principals.
+---
+
+# Auth Backend: aws-iam
+
+The aws-iam auth backend provides a secure introduction mechanism for AWS IAM
+principals. This allows you to reduce the problem of securely introducing a
+Vault token to the problem of securely introducing AWS IAM credentials, which
+AWS has already solved in a number of use cases, such as via EC2 Instance
+Profiles and IAM roles attached to Lambda functions. It also allows you to have
+a consistent workflow between developers working on a local laptop with tools
+such as Hologram.
+
+## Comparison with AWS-EC2 Authentication Backend
+
+The AWS-IAM and AWS-EC2 authentication backends serve similar purposes. Both
```

Comment:
@jantman -- it depends on what your threat model is. If your threat model is malicious EC2 admins attempting to escalate their privileges, then limiting based on VPC isn't a strong control since the AWS IAM language is a bit awkward in trying to limit people to be able to only deploy inside of a VPC or subnet (they've gotten better about allowing you to specify ec2:Vpc and/or ec2:Subnet inside of condition keys, but IIRC, that wasn't always the case). And trying to limit based on the role_tag passed in is, as far as I can tell, impossible from an IAM language standpoint right now.

If you trust your EC2 admins to only launch instances into the right VPC using the right AMI and the right tags, and your main threat model is more about a machine getting compromised and you want to limit the blast radius of an attack, then limiting based off of VPC could indeed help out a lot there. I don't see any reason in theory why you couldn't just create unique IAM instance profiles and roles for each VPC or subnet (per your example) and even just attach the same managed policy to each of the roles, and achieve the same thing (unless you have existing deployments where switching to this would be difficult and not worth the cost).

FWIW, my own _personal_ opinion is that launching instances into an instance profile is treated as a much more security sensitive operation than launching an instance into a particular VPC or subnet as the former is a credential-granting operation and potential privilege escalation vector, while the latter isn't really treated as such. So you'd be much better off just creating unique IAM roles for each separate Vault role you want to grant your EC2 instances access to, and then protecting iam:PassRole access to each role accordingly. But, again, that's just my personal opinion, and my personal biases, which I'm trying to avoid in writing the documentation.